sdk/storage/azure-storage-internal-avro/src/main/java/com/azure/storage/internal/avro/implementation/AvroParser.java

```diff
@@ -106,31 +103,26 @@ private void onFilteredHeader(Object header) {
 
         /* On reading the header, read a block. */
         if (!partialRead) { /* Only do this if we are reading the stream from start to finish. */
-            onBlock(-1L);
+            onBlock(0L);
         }
     }
 
     /**
      * Block handler.
      *
-     * @param index The object index after which to start aggregating events in the block.
+     * @param threshold The object index after which to start aggregating events in the block.
      *                         By default this is 0 to collect all objects in the block.
      */
-    private void onBlock(Object index) {
+    private void onBlock(Object threshold) {
         /* On reading the block, read another block. */
-        AvroSchema.checkType("objectBlockIndex", index, Long.class);
-        long threshold = (Long) index;
+        AvroSchema.checkType("threshold", threshold, Long.class);
 
-        this.blockOffset = this.state.getSourceOffset();
-        this.objectBlockIndex = 0;
-        AvroBlockSchema blockSchema = new AvroBlockSchema(
+        final AvroBlockSchema blockSchema = new AvroBlockSchema(
             this.objectType,
+            (Long) threshold,
```

Comment:
maybe this should be called beginIndex or beginObjectIndex. Threshold sounds more like upper boundary.

Reply:
done